music/src/core/compat/global_browser.ts

```diff
@@ -18,6 +18,42 @@
  * limitations under the License.
  */
 
-export const fetch = window.fetch.bind(window);
-export const performance = window.performance;
-export const navigator = window.navigator;
+/**
+ * Attempts to determine which global object to use, depending on whether or
+ * note the script is being used in a browser, web worker, or other context.
+ */
+function getGlobalObject() {
+  // tslint:disable-next-line:max-line-length
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis
+  if (typeof globalThis !== 'undefined') { return globalThis; }
+  if (typeof self !== 'undefined') { return self; }
+  if (typeof window !== 'undefined') { return window; }
+  if (typeof global !== 'undefined') { return global; }
+  throw new Error('cannot find the global object');
+}
+
+const globalObject = getGlobalObject();
+
+export const fetch = globalObject.fetch.bind(globalObject);
+export const performance = globalObject.performance;
+export const navigator = globalObject.navigator;
+
+export function isSafari() {
+  return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
+}
+
+function isWorker() {
+  // tslint:disable-next-line:no-any variable-name
+  return typeof (globalObject as any).WorkerGlobalScope !== 'undefined';
+}
+
+export function getOfflineAudioContext(sampleRate: number): OfflineAudioContext {
+  if (isWorker()) {
+    throw new Error('Cannot use offline audio context in a web worker.');
+  }
+
+  // tslint:disable-next-line:no-any variable-name
+  const SafariOfflineCtx = (globalObject as any).webkitOfflineAudioContext;
+  return isSafari() ? new SafariOfflineCtx(1, sampleRate, sampleRate) :
```

Comment:
This check is technically incorrect, but if you've tested it in Safari/Chrome on iOS etc, firefox on iOS etc it should be fine. IMO you should be feature checking whether or not `webkitOfflineAudioContext` exists, since that's what you're trying to use, rather than using the user agent string at all. I think the best practices around user agent strings is "use them if you can't find a feature to check", which in this case we have

Reply:
Yup